internal/dataplane/kongstate/kongstate_test.go

```diff
@@ -335,64 +346,226 @@ func TestFillConsumersAndCredentials(t *testing.T) {
+						},
+					},
+					Username: "foo",
+					Credentials: []string{
+						"nonExistCredSecret",
+					},
+				},
+			},
+			expectedKongStateConsumers: []Consumer{
 				{
-					kong.Oauth2Credential{
-						Name:         kong.String("whatever"),
-						ClientID:     kong.String("whatever"),
-						ClientSecret: kong.String("whatever"),
-						HashSecret:   kong.Bool(true),
-						RedirectURIs: []*string{kong.String("http://example.com")},
+					Consumer: kong.Consumer{
+						Username: kong.String("foo"),
 					},
 				},
 			},
-		}},
+			expectedTranslationFailures: map[k8stypes.NamespacedName]failures.ResourceFailure{
+				{Namespace: "default", Name: "foo"}: mustNewResourceFailure(t,
+					"failed to fetch secret",
+					&configurationv1.KongConsumer{
+						TypeMeta:   kongConsumerTypeMeta,
+						ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "foo"},
+					}),
+			},
+		},
+		{
+			name: "referring to secret with unsupported credType",
+			k8sConsumers: []*configurationv1.KongConsumer{
+				{
+					TypeMeta: kongConsumerTypeMeta,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "foo",
+						Namespace: "default",
+						Annotations: map[string]string{
+							"kubernetes.io/ingress.class": annotations.DefaultIngressClass,
+						},
+					},
+					Username: "foo",
+					Credentials: []string{
+						"unsupportedCredSecret",
+					},
+				},
+			},
+			expectedKongStateConsumers: []Consumer{
+				{
+					Consumer: kong.Consumer{
+						Username: kong.String("foo"),
+					},
+				},
+			},
+			expectedTranslationFailures: map[k8stypes.NamespacedName]failures.ResourceFailure{
+				{Namespace: "default", Name: "foo"}: mustNewResourceFailure(t,
+					"failed to provision credential: invalid credType: unsupported",
+					&configurationv1.KongConsumer{
+						TypeMeta:   kongConsumerTypeMeta,
+						ObjectMeta: metav1.ObjectMeta{Namespace: "default", Name: "foo"},
+					}),
+			},
+		},
 	}
 
-	t.Run("parses consumer and credential from store into state", func(t *testing.T) {
-		state := KongState{}
-		state.FillConsumersAndCredentials(logrus.New(), store, semver.MustParse("2.3.2"))
-		assert.Equal(t, want.Consumers[0].Consumer.Username, state.Consumers[0].Consumer.Username)
-		assert.Equal(t, want.Consumers[0].Consumer.CustomID, state.Consumers[0].Consumer.CustomID)
-		assert.Equal(t, want.Consumers[0].KeyAuths[0].Key, state.Consumers[0].KeyAuths[0].Key)
-		assert.Equal(t, want.Consumers[0].Oauth2Creds[0].ClientID, state.Consumers[0].Oauth2Creds[0].ClientID)
-		assert.Equal(t, want.Consumers[0].Oauth2Creds[0].ClientSecret, state.Consumers[0].Oauth2Creds[0].ClientSecret)
-		assert.Equal(t, want.Consumers[0].Oauth2Creds[0].HashSecret, state.Consumers[0].Oauth2Creds[0].HashSecret)
-		assert.Equal(t, want.Consumers[0].Oauth2Creds[0].RedirectURIs, state.Consumers[0].Oauth2Creds[0].RedirectURIs)
-	})
+	for i, tc := range testCases {
+		indexStr := strconv.Itoa(i)
+		tc := tc
+		t.Run(indexStr+"-"+tc.name, func(t *testing.T) {
```

Comment:
Why not just a name? They are unique, aren't they?
```suggestion
		tc := tc
		t.Run(tc.name, func(t *testing.T) {
```

Reply:
I just wanted to print the index of test case when test fails to find it quickly, because sometimes I will fold the long long struct in the editor and the name is not displayed in such situation.

Reply:
Hm, I see. In GoLand when the test fails I can easily click on the failed test and it will take me to the test case in code. I see your point that it may be quicker this way.

BTW IMO we've made a bad choice by using test names with spaces - that prevents us from just copying the test case name from the output and finding it directly in the code. By using spaces, `go test`'s output gives us test names with spaces transformed into `_`. I tended to use underscores instead of spaces in test cases name in the past and it didn't hurt that much. :)